### PR TITLE
Do not set _spec if an error is raised.

### PR DESCRIPTION
**Pull request by greschd**
When an error is raised while building the spec of a process, the ``_spec`` attribute is deleted, and ``__called`` set to ``False``. This ensures that subsequent calls to ``.spec()`` also raise the error instead of returning an incomplete spec.

Related AiiDA issue and PR: https://github.com/aiidateam/aiida-core/issues/3604, https://github.com/aiidateam/aiida-core/pull/3617

### CODE REVIEW

File: test/test_processes.py [hunk 1]
```diff
@@ -143,6 +143,23 @@ def define(cls, spec):
...
+                super(BrokenProcess, cls).define(spec)
+                raise ValueError
+
+        with self.assertRaises(ValueError):
+            BrokenProcess.spec()
+        # Check that the error is still raised when calling .spec()
+        # a second time.
+        with self.assertRaises(ValueError):
+            BrokenProcess.spec()
+        self.assertFalse(hasattr(BrokenProcess, '_spec'))
```
Comment (muhrin) [inline test/test_processes.py]: Could you remove this check?  I'm not a big fan of unit test checking implementation details.

**Comment by muhrin** [comment]
Just to keep you updated: I'm working on fixing the tests which seem to have broken because of a new version of PyYaml

File: test/test_processes.py (hunk 1 above)
Comment (greschd) [reply test/test_processes.py]: Done!

Comment (muhrin) [reply test/test_processes.py]: Thanks!  Now just waiting for PRs on `topika` and `kiwipy` so I can release new versions and fix these tests.

Comment (greschd) [reply test/test_processes.py]: It's PRs all the way down 😄

**Pull request merged**